src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +36,128 @@ const ActorInit Bg_Jya_Lift_InitVars = {
+}
+
+
+void BgJyaLiftSetInitPosY(BgJyaLift* this) {
+    this->actionFunc = func_80899D38;
+    this->dyna.actor.posRot.pos.y = 1613.0f;
+    this->unk_168 = 0;
+}
+
+
+void func_80899D38(BgJyaLift* this, GlobalContext* globalCtx) {
+    if ((Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F)) != 0) || (this->unk_168 > 0)) {
+        this->unk_168 = (this->unk_168 + 1);
+        if (this->unk_168 >= 0x14) {
+            func_800800F8(globalCtx, 0xD66, -0x63, &this->dyna.actor, 0);
+            func_80899DBC(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Init.s")
+void func_80899DBC(BgJyaLift* this) {
+    this->actionFunc = BgJyaLift_Move;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Destroy.s")
+void BgJyaLift_Move(BgJyaLift* this, GlobalContext* globalCtx) {
+    f32 distFromBottom;
+    f32 tempVelocity;
+    Math_SmoothScaleMaxMinF(&this->dyna.actor.velocity.y, 4.0f, 0.1f, 1.0f, 0.0f);
+    tempVelocity = (this->dyna.actor.velocity.y < 0.2f) ? 0.2f : this->dyna.actor.velocity.y;
+    distFromBottom = Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, 973.0f, 0.1f, tempVelocity, 0.2f); 
+    
+    if ((this->dyna.actor.posRot.pos.y < 1440.0f) && (1440.0f <= this->dyna.actor.pos4.y)) {
+        func_8005B1A4(ACTIVE_CAM);
+    }
+    if (fabsf(distFromBottom) < 0.001f) { 
+        BgJyaLiftSetFinalPosY(this);
+        Audio_PlayActorSound2(&this->dyna.actor, 0x287A);
+        return;
+    } else {
+        func_8002F974((Actor*)this, 0x200C);
+     }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D18.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D38.s")
+void BgJyaLiftSetFinalPosY(BgJyaLift* this) {
+    this->actionFunc = 0;
+    this->dyna.actor.posRot.pos.y = 973.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899DBC.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899DCC.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899EF0.s")
+void BgJyaLift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
+    if (this->actionFunc) {
+        (this->actionFunc)(this);
+    }
+    if ((((this->dyna.unk_160) & 4) != 0) && ((this->unk_16B & 4) == 0)) {
+        func_8005A77C(globalCtx2->cameraPtrs[0], 0x3F);
+    } else {
+        if (((this->dyna.unk_160) & 4) == 0 && ((this->unk_16B & 4) != 0) && (globalCtx2->cameraPtrs[0]->setting == 0x3F)) {
+        func_8005A77C(globalCtx2->cameraPtrs[0], 3);
+        }
+    }
+    this->unk_16B = this->dyna.unk_160;
+    if ((globalCtx2->roomCtx.curRoom.num != 5) && (globalCtx2->roomCtx.curRoom.num != 25)) { 
+            Actor_Kill(thisx);
+       // Spirit Temple room 5 is the main room with the statue room 25 is directly above room 5
```

Comment:
id put the comment on top of the if, with maybe a line break to separate it from the code above it